download-episodes.py

```diff
@@ -0,0 +1,237 @@
+        file_data = f.read()
+        return json.loads(file_data)
+
+
+def _get_show_response():
+    if args.with_fake_response:
+        channel_response = _read_catchup_response_from_file()
+    else:
+        channel_response = _get_channel_catchup_response()
+
+    for show in channel_response:
+        if show['showId'] == show_config['show_id']:
+            return show
+
+
+def _get_file_format():
+    if show_config['file_format']:
+        return show_config['file_format']
+    else:
+        return 'm4a'
+
+
+def _get_episode_downloads_folder():
+    if show_config['download_folder']:
+        download_folder = show_config['download_folder']
+    else:
+        download_folder = DOWNLOAD_PATH
+
+    download_folder = download_folder.strip()
+
+    os.makedirs(name=os.path.expanduser(download_folder), exist_ok=True)
+
+    if download_folder.endswith("/"):
+        return download_folder
+    else:
+        return download_folder + "/"
+
+
+def get_downloaded_episodes():
+    verbose('Getting list of all downloaded episodes...')
+    download_folder = os.path.expanduser(_get_episode_downloads_folder())
+    os.chdir(download_folder)
+
+    files = glob('*.' + _get_file_format())
+
+    verbose(f'Found {len(files)} downloads in the folder...')
+    downloads = []
+    for file in files:
+        downloaded_file = Download(file)
+        downloads.append(downloaded_file)
+
+    return downloads
+
+
+def _parse_episodes(show):
+    parsed_episodes = []
+    for episode in show['episodes']:
+        parsed = Episode(episode)
+        parsed_episodes.append(parsed)
+
+    return parsed_episodes
+
+
+def get_latest_episodes():
+    show = _get_show_response()
+    verbose(f'Show json response: {json.dumps(show, indent=2)}')
+    episodes = _parse_episodes(show)
+    verbose(f'Found {len(episodes)} episodes in show response...')
+    for e in episodes:
+        e.print()
+    return episodes
+
+
+def download(episode):
+    if args.with_fake_response:
+        progress_complete(episode.title_with_date, 100)
+        return
+
+    url_request = request.Request(episode.stream_url)
+    url_connect = request.urlopen(url_request)
+
+    file_size = int(url_connect.info()['Content-Length'])
+    download_block_size = 8192
+    downloaded_size = 0
+
+    out_file_name = datetime.strftime(episode.date, "%Y%m%d") + '.' + _get_file_format()
+    download_folder = os.path.expanduser(_get_episode_downloads_folder())
+    tmp_folder = download_folder + 'tmp'
+
+    if not os.path.exists(tmp_folder):
+        os.mkdir(tmp_folder)
+    os.chdir(tmp_folder)
+
+    with open(out_file_name, 'wb') as out_file:
+        while True:
+            buffer = url_connect.read(download_block_size)
+            downloaded_size += len(buffer)
+            if not buffer:
+                break
+            out_file.write(buffer)
+            progress(episode.title_with_date, file_size, downloaded_size)
+
+    os.rename(out_file_name, download_folder + out_file_name)
```

Comment:
You can use `os.path.join(download_folder, out_file_name)` btw.  Should be a bit more cross-platform as that knows which separators to use etc.